### PR TITLE
[tests-only]Add tests for renaming a file after public link share

### PR DESCRIPTION
**Pull request by SwikritiT**
## Description
Bug demonstration for : https://github.com/owncloud/core/issues/39820

## Related Issue

- Part of: https://github.com/owncloud/core/issues/39820
- https://github.com/owncloud/ocis/issues/2915
## How Has This Been Tested?

- locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature [hunk 1]
```diff
@@ -103,14 +103,16 @@ Feature: update a public link share
...
+      | path                       | /RENAMED_FOLDER      |
       | mimetype                   | httpd/unix-directory |
-      | storage_id                 |                      |
-      | storage                    | 0                    |
-      | file_source                |                      |
-      | file_target                |                      |
+      | storage_id                 | A_STRING             |
+      | storage                    | A_STRING             |
+      | file_source                | A_STRING             |
+      # uncomment the following like and remove the next one after the issue has been fixed
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature:113]:
```suggestion
      # uncomment the following line and remove the next one after the issue has been fixed
```

File: tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature [hunk 2]
```diff
@@ -499,3 +495,139 @@ Feature: update a public link share
       | ocs_api_version | ocs_status_code | webdav_api_version |
       | 1               | 100             | new                |
       | 2               | 200             | new                |
+
+  @skipOnOcV10
+  Scenario Outline: API responds with a full set of parameters when owner renames the folder with a public link in ocis
+    Given using OCS API version "<ocs_api_version>"
+    And using <dav-path> DAV path
+    And user "Alice" has created folder "FOLDER"
+    When user "Alice" creates a public link share using the sharing API with settings
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature:504]: seems like a given step.

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIUpload-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34729/151/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIWebdavLockProt-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34729/152/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIWebdavLocks-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34729/153/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingNotify-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34729/154/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUISharingExt1-latest-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34729/157/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUIFileActionsMenu-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34729/155/1

**Comment by ownclouders** [comment]
:boom: Acceptance tests pipeline <strong>webUITrashbin-chrome-mariadb10.2-php7.4</strong> failed. The build has been cancelled.

https://drone.owncloud.com/owncloud/core/34729/150/1

File: tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature [hunk 3]
```diff
@@ -499,3 +495,139 @@ Feature: update a public link share
...
+      | additional_info_owner      |                |
+      | additional_info_file_owner |                |
+      | item_type                  | file           |
+      | item_source                | A_STRING       |
+      | path                       | /new-lorem.txt |
+      | mimetype                   | text/plain     |
+      | storage_id                 | A_STRING       |
+      | storage                    | A_STRING       |
+      | file_source                | A_STRING       |
+      # uncomment the following like and remove the next one after the issue has been fixed
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature:577]:
```suggestion
      # uncomment the following line and remove the next one after the issue has been fixed
```

File: tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature [hunk 4]
```diff
@@ -499,3 +495,139 @@ Feature: update a public link share
...
+      | old      | 2               | 200             |
+      | new      | 1               | 100             |
+      | new      | 2               | 200             |
+
+  @skipOnOcV10
+  Scenario Outline: API responds with a full set of parameters when owner renames the file with a public link in ocis
+    Given using OCS API version "<ocs_api_version>"
+    And using <dav-path> DAV path
+    Given user "Alice" has uploaded file with content "some content" to "/lorem.txt"
+    When user "Alice" creates a public link share using the sharing API with settings
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature:594]: here too. for a renaming scenario, create share part can go to `given` step

File: tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature [hunk 5]
```diff
@@ -499,3 +495,139 @@ Feature: update a public link share
...
+      | old      | 1               | 100             |
+      | old      | 2               | 200             |
+      | new      | 1               | 100             |
+      | new      | 2               | 200             |
+
+  @skipOnOcV10
+  Scenario Outline: API responds with a full set of parameters when owner renames the file with a public link in ocis
+    Given using OCS API version "<ocs_api_version>"
+    And using <dav-path> DAV path
+    Given user "Alice" has uploaded file with content "some content" to "/lorem.txt"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature:593]:
```suggestion
    And user "Alice" has uploaded file with content "some content" to "/lorem.txt"
```

File: tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature [hunk 6]
```diff
@@ -499,3 +495,139 @@ Feature: update a public link share
...
+    Examples:
+      | dav-path | ocs_api_version | ocs_status_code |
+      | spaces   | 1               | 100             |
+      | spaces   | 2               | 200             |
+
+  @notToImplementOnOCIS @issue-39820
+  Scenario Outline: API responds with a full set of parameters when owner renames the file with a public link (bug demonstration)
+    Given using OCS API version "<ocs_api_version>"
+    And using <dav-path> DAV path
+    Given user "Alice" has uploaded file with content "some content" to "/lorem.txt"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiSharePublicLink2/updatePublicLinkShare.feature:549]:
```suggestion
    And user "Alice" has uploaded file with content "some content" to "/lorem.txt"
```

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by saw-jan — COMMENTED** [review]
LGTM

**Review by phil-davis — APPROVED** [review]

**Pull request merged**